src/main/java/com/epam/prejap/teatrees/game/Playfield.java

```diff
@@ -99,12 +102,26 @@ private void forEachBrick(BrickAction action) {
             for (int j = 0; j < block.cols(); j++) {
                 var dot = block.dotAt(i, j);
                 if (dot > 0) {
-                    action.act(i, j, dot);
+                    action.act(i, j, differentiator);
                 }
             }
         }
     }
 
+    /**
+     * Loops the color indicator value named in the method as a <i>differentiator</i>.
+     * <p>When the value exceeds the size of the color palette (AnsiColors),
+     * the indicator is reset to zero.</p>
+     *
+     * @see AnsiColors
+     */
+    private void updateDifferentiator() {
+        differentiator++;
+        if (differentiator > AnsiColors.values().length) {
```

Comment:
Maye be simplified IIRC to 

```suggestion
        if (++differentiator > AnsiColors.values().length) {
```

Pre-incrementation isnted of post-incerementation

Reply:
done